README.md

```diff
@@ -1,2 +1,69 @@
-# rustls-symcrypt
-Glue between Rustls and rust-symcrypt 
+# SymCrypt Provider for Rustls
+
+This crate provides integration for using [rust-symcrypt](https://github.com/microsoft/rust-symcrypt) cryptographic functionalities with  [rustls](https://github.com/rustls/rustls), by implementing the required traits specified by `rustls`.
+
+## Status and Platform Support:
+- Currently works with: `rustls = { version = "0.23.0", features = ["ring", "tls12", "std"], default-features = false }`
+- Windows AMD64: Full support.
+- Linux Mariner: Full support.
+- Ubuntu: Partial support. While tested, full compatibility and optimal performance on all Ubuntu environments cannot be guaranteed.
+
+## Dependencies
+
+This crate depends on the [symcrypt](https://github.com/microsoft/rust-symcrypt) crate and requires you have the necessary `symcrypt` binaries for your architecture.
+Refer to the [rust-symcrypt Quick Start Guide](https://github.com/microsoft/rust-symcrypt/tree/main/rust-symcrypt#quick-start-guide) to download the required binaries.
+
+## Usage
+
+Add `rustls-symcrypt` to your `Cargo.toml`:
+**Note:** If you wish to enable `x25519` or `chacha` you may add it as a feature at this time.
+
+```toml
+[dependencies]
+rustls = { version = "0.23.0", features = ["ring", "tls12", "std"], default-features = false }
+# Disabling aws-lc as it slows down build times and is not needed.
+rustls_symcrypt = "0.1.0"
+# To enable the chacha feature:
+# rustls_symcrypt = {version = "0.1.0", features = ["chacha"]}
+```
+
+## Supported Ciphers
+
+Supported cipher suites are listed below, ordered by preference. IE: The default configuration prioritizes `TLS13_AES_256_GCM_SHA384` over `TLS13_AES_128_GCM_SHA256`.
+
+### TLS 1.3
+
+```ignore
+TLS13_AES_256_GCM_SHA384
+TLS13_AES_128_GCM_SHA256
+TLS13_CHACHA20_POLY1305_SHA256 // Enabled with the `chacha` feature
+```
+
+**Note:** `TLS13_CHACHA20_POLY1305_SHA256` is disabled by default. Enable the `chacha` feature in your `Cargo.toml` to use this cipher suite.
+
+### TLS 1.2
+
+```ignore
+TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
```

Comment:
Is AES-128 preferred over AES-256 in TLS 1.2?

Reply:
It is not, nice catch thanks